diff --git a/glib/glib-unix.c b/glib/glib-unix.c
index 932c230..8b1fcaf 100644
--- a/glib/glib-unix.c
+++ b/glib/glib-unix.c
@@ -77,7 +77,7 @@ G_STATIC_ASSERT (G_ALIGNOF (GPid) == G_ALIGNOF (pid_t));
  * ambiguous on this platform.
  * On Linux, usually O_NONBLOCK == 04000 and FD_CLOEXEC == 1, but the same
  * might not be true everywhere. */
-G_STATIC_ASSERT (O_NONBLOCK != FD_CLOEXEC);
+/* cosmopolitan: O_NONBLOCK is not a compile-time constant */
 
 G_DEFINE_QUARK (g-unix-error-quark, g_unix_error)
 
