--- a/include/qemu/osdep.h
+++ b/include/qemu/osdep.h
@@ -114,6 +114,9 @@
 #include <stdio.h>
 
 #include <string.h>
+#ifdef __COSMOPOLITAN__
+#include "qemu/cosmo-sscanf.h"
+#endif
 #include <strings.h>
 #include <inttypes.h>
 #include <limits.h>
--- a/util/meson.build
+++ b/util/meson.build
@@ -1,4 +1,5 @@
 util_ss.add(files('osdep.c', 'cutils.c', 'unicode.c', 'qemu-timer-common.c'))
+util_ss.add(files('cosmo-sscanf.c'))
 util_ss.add(files('thread-context.c'), numa)
 if not config_host_data.get('CONFIG_ATOMIC64')
   util_ss.add(files('atomic64.c'))
