diff --git a/block/file-posix.c b/block/file-posix.c
index 90fa543..06e418e 100644
--- a/block/file-posix.c
+++ b/block/file-posix.c
@@ -1062,8 +1062,9 @@ static int raw_reconfigure_getfd(BlockDriverState *bs, int flags,
 
     raw_parse_flags(flags, open_flags, has_writers);
 
-#ifdef O_ASYNC
-    /* Not all operating systems have O_ASYNC, and those that don't
+#if defined(O_ASYNC) && !defined(__COSMOPOLITAN__)
+    /* (cosmopolitan defines flags a system lacks as all ones, not zero)
+     * Not all operating systems have O_ASYNC, and those that don't
      * will not let us track the state into rs->open_flags (typically
      * you achieve the same effect with an ioctl, for example I_SETSIG
      * on Solaris). But we do not use O_ASYNC, so that's fine.
